plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/repository/LibraryClasspathContainerResolverService.java

```diff
@@ -138,7 +138,33 @@ public IStatus resolveContainer(IJavaProject javaProject, IPath containerPath,
       return StatusUtil.error(this, Messages.getString("TaskResolveContainerError", containerPath), ex);
     }
   }
-  
+
+  public IStatus checkRuntimeAvailability(Runtime runtime, IProgressMonitor monitor) {
+    switch (runtime) {
+      case AppEngineStandard:
+      return checkAppEngineStandard(monitor);
+      default:
+        throw new IllegalArgumentException("Unhandled runtime: " + runtime);
+      }
+  }
+
+  private IStatus checkAppEngineStandard(IProgressMonitor monitor) {
```

Comment:
I'd argue that it's the other way around: our GAE Standard environment happens to be made up of servlet and the jsp jars, but if it contained another jar, that would also be checked here. Also the version, although not explicitly mentioned her (servlet 2.5, jsp 2.1) is also important part of the check.